lighthouse-core/gather/driver.js

```diff
@@ -570,9 +570,14 @@ class Driver {
       });
   }
 
-  beginTrace() {
+  /**
+   * @param {{traceCategories: string=}=} flags
+   */
+  beginTrace(flags) {
+    const additionalCategories = flags && flags.traceCategories && flags.traceCategories.split(',');
+    const traceCategories = new Set(this._traceCategories.concat(additionalCategories || []));
```

Comment:
i kinda prefer moving this ` || []` to the above statement, so we can rely on it as an array.